src/utils/fetchCreateTokenEvents.ts

```diff
@@ -36,53 +36,58 @@ async function findBlock(blockNumber: number): Promise<BlockI> {
 }
 
 export async function getTokenCreatedEvents() {
-  logger.info('==== Start event import ====')
-  const contract = new ethers.Contract(contractAddress, abi, provider)
+  try {
+    logger.info('==== Start event import ====')
+    const contract = new ethers.Contract(contractAddress, abi, provider)
 
-  const filter = 'TokenCreated'
-  const startBlock = await getLatestEventBlockNumberFromDb()
-  const endBlock = 'latest'
-  const events = await contract.queryFilter(filter, startBlock, endBlock)
-  const cleanedEvents: CreateTokenEventI[] = []
+    const filter = 'TokenCreated'
+    const startBlock = (await getLatestEventBlockNumberFromDb()) ?? 0
+    const endBlock = 'latest'
+
+    const events = await contract.queryFilter(filter, startBlock, endBlock)
+    const cleanedEvents: CreateTokenEventI[] = []
+
+    for (const event of events) {
+      const { blockNumber, transactionHash }: { blockNumber: number; transactionHash: string } = event
+      const existingEvent = await findCreateTokenEvent(transactionHash)
+      const eventBlock = await findBlock(blockNumber)
+      if (!eventBlock && !existingEvent) {
+        cleanedEvents.push({
+          blockNumber,
+          unixTimestamp: 1234567890,
+          timestamp: getDateFromUnixTimestamp(1234567890),
+          transactionHash
+        })
+        logger.info(`Added TokenCreated event: ${transactionHash} Block: ${blockNumber}`)
+        continue
+      }
+      if (!eventBlock) {
+        continue
+      }
```

Comment:
suggestion(blocking): you could save a check if you check this before the previous if (including a `existingEvent` check). Removing one level of indentation.